viewer/viewer.js

```diff
@@ -657,6 +657,34 @@ function arrayZeroFill(n) {
   return a;
 }
 
+// https://stackoverflow.com/a/48569020
+class Mutex {
+    constructor () {
+        this.queue = [];
+        this.locked = false;
+    }
+
+    lock () {
+        return new Promise((resolve, reject) => {
+            if (this.locked) {
+                this.queue.push([resolve, reject]);
+            } else {
+                this.locked = true;
+                resolve();
+            }
+        });
+    }
+
+    unlock () {
+        if (this.queue.length > 0) {
+            const [resolve, reject] = this.queue.shift();
```

Comment:
Is reject necessary here? This will fail linting cause it is an unused variable.